### PR TITLE
Fix OPL3

### LINKED ISSUE #74: OPL3 class seems to lack OPL3-specific addressing
I previously wrote some code to do MIDI-to-OPL2 and play old tunes through a MIDI console. I recently upgraded to (single) OPL3 using the OPL3Duo, which sounded great as a drop-in replacement. To level up my game, I enabled 4-op mode, but I got a feeling I could not achieve full 6-voice polyphony: notes sounded like they were getting "stolen" by my voice allocator.

I did some debugging, and the voice allocator works fine. Actually, it did before on 6-voice 2-op OPL2, so I dug a little deeper. My next guess was there was something wrong with the BANK/A1 pin, which would effectively halve the amount of voices. And also, it would prevent the chip from going into OPL3 mode. I started comparing the sounds to an emulated OPL3, and it could be the case.

Looking at the code:
* OPL3.cpp does not have its own `setChipRegister`, so defaults to the OPL2 implementation, which masks the register with 0xff. Registers 0x100 and up can never be addressed in this way
* OPL3.cpp does have a `setChannelRegister` and `setOperatorRegister`, but BANK/A1 only get driven when `(channel >> 8) & 0x01`, which is of course never the case

I am trying to get my head around the datasheet/programming guide to come up with a PR to address these issues, but if anyone is more fluent in YM262, I would appreciate some help :-)

### PR DESCRIPTION
**Pull request by jonkerj**
It seems the OPL3-specific code is not fully able to use the OPL3 features at all. This PR fixes this. Tested on Arduino with OPL3.

Fixes #74

### CODE REVIEW

File: src/OPL3.cpp [hunk 1]
```diff
@@ -139,17 +139,32 @@ byte OPL3::getChipRegisterOffset(short reg) {
 }
 
 
+/**
+ * Write a given value to a chip wide register.
+ *
+ * @param synthUnit - The chip to address [0, 1]
```
Comment (DhrBaksteen) [inline src/OPL3.cpp:145]: This param should be removed since the base OPL3 class is not handling multiple chips.

File: src/OPL3.cpp [hunk 2]
```diff
@@ -139,17 +139,32 @@ byte OPL3::getChipRegisterOffset(short reg) {
...
+
 /**
  * Write a given value to a channel based register.
  *
  * @param baseRegister - The base address of the register.
  * @param channel - The channel to address [0, 17]
  * @param value - The value to write to the register.
  */
-void OPL3::setChannelRegister(byte baseRegister, byte channel, byte value) {
+void OPL3::setChannelRegister(short baseRegister, byte channel, byte value) {
```
Comment (DhrBaksteen) [inline src/OPL3.cpp:164]: Using a `short` here could give the impression that the `baseRegister` is also used to select between bank 0 or 1, while this is determined by the `channel`. The `baseRegister` for a channel can only be 0xA0, 0xB0 or 0xC0.

File: src/OPL3.cpp [hunk 3]
```diff
@@ -163,10 +178,10 @@ void OPL3::setChannelRegister(byte baseRegister, byte channel, byte value) {
  * @param operatorNum - The operator to change [0, 1].
  * @param value - The value to write to the operator's register.
  */
-void OPL3::setOperatorRegister(byte baseRegister, byte channel, byte operatorNum, byte value) {
+void OPL3::setOperatorRegister(short baseRegister, byte channel, byte operatorNum, byte value) {
```
Comment (DhrBaksteen) [inline src/OPL3.cpp:181]: Same here.

File: src/OPL3.cpp (hunk 1 above)
Comment (jonkerj) [reply src/OPL3.cpp]: Oops, I was a bit too liberal with copy pasting from OPL3Duo

File: src/OPL3.cpp (hunk 2 above)
Comment (jonkerj) [reply src/OPL3.cpp]: same here, I borrowed it from the other class. Fixed in new force-pushed commit

File: src/OPL3.cpp (hunk 3 above)
Comment (jonkerj) [reply src/OPL3.cpp]: same here, I borrowed it from the other class. Fixed in new force-pushed commit

**Review by DhrBaksteen — APPROVED** [review]
Tested good. Thanks a lot for your contribution @jonkerj!

**Pull request merged**